tests/system/test_misc.c

```diff
@@ -389,3 +389,88 @@ int cmd_exit_cb(int argc, char **argv)
    DEVSHELL_CMD_ASSERT(after_cb == before_cb + 1);
    return 0;
 }
+
+static bool
+comes_after(struct timeval t2,
+            struct timeval t1)
+{
+   // t1 must come before t2
+   return t2.tv_sec > t1.tv_sec || (t2.tv_sec == t1.tv_sec && t2.tv_usec > t1.tv_usec);
+}
+
+#define BEGIN_OF_TIME (struct timeval) {0, 0}
+
+static void busy_wait(void)
+{
+   // It's assumed that this loop takes
+   // more than 1us of cpu time.
+   for (int i = 0; i < 1000000; i++)
+      asm(""); // Don't let the compiler optimize this loop
+}
+
+static void busy_wait_kernel(void)
+{
+   // It's assumed that this loop takes
+   // more than 1us of cpu time in kernel
+   // space.
+   for (int i = 0; i < 100; i++) {
+      int n = fork();
+      if (n == 0)
+         exit(0);
+   }
+}
+
+int cmd_getrusage(int argc, char **argv)
+{
+   int rc;
+   struct rusage buf_0;
+   struct rusage buf_1;
+
+   // Do some work to make sure that the user cpu time
+   // isn't less than 1us
+   busy_wait();
+   busy_wait_kernel();
+
+   // Check that RUSAGE_SELF doesn't fail and that
+   // the returned utime and stime aren't 0.
+   rc = syscall(SYS_getrusage, RUSAGE_SELF, &buf_0);
```

Comment:
Don't use `syscall()` unless it's really necessary. Use the libc wrapper instead.